aws/rust-runtime/aws-types/src/sdk_config.rs

```diff
@@ -359,15 +359,84 @@ impl Builder {
         self
     }
 
-    /// Sets the HTTP connector that clients will use to make HTTP requests.
-    pub fn http_connector(mut self, http_connector: HttpConnector) -> Self {
+    /// Sets the HTTP connector to use when making requests.
+    ///
+    /// ## Examples
+    /// ```
+    /// # #[cfg(test)]
+    /// # mod tests {
+    /// # #[test]
+    /// # fn example() {
+    /// use std::time::Duration;
+    /// use aws_smithy_client::{Client, hyper_ext};
+    /// use aws_smithy_client::erase::DynConnector;
+    /// use aws_smithy_client::http_connector::ConnectorSettings;
+    /// use aws_types::SdkConfig;
+    ///
+    /// let https_connector = hyper_rustls::HttpsConnectorBuilder::new()
+    ///     .with_webpki_roots()
+    ///     .https_only()
+    ///     .enable_http1()
+    ///     .enable_http2()
+    ///     .build();
+    /// let smithy_connector = hyper_ext::Adapter::builder()
+    ///     // Optionally set things like timeouts as well
+    ///     .connector_settings(
+    ///         ConnectorSettings::builder()
+    ///             .connect_timeout(Duration::from_secs(5))
+    ///             .build()
+    ///     )
+    ///     .build(https_connector);
+    /// let sdk_config = SdkConfig::builder()
+    ///     .http_connector(smithy_connector)
+    ///     .build();
```

Comment:
I think `SdkConfig::builder` is the wrong example to be setting, but it's not easy to include `aws-config` in the examples... Maybe we can generate a Rust code snippet as a separate file that recreates a minimal `ConfigLoader` just for examples, and include that (hidden) in each doc example that needs to illustrate it? Or just set these to `no_compile`, but that kind of stinks.

Or maybe we somehow refactor `ConfigLoader` into `aws-types` with the entry point `aws_config::from_env` returning a concrete implementation?